src/test_l1_dsFPD.c

```diff
@@ -1371,21 +1588,36 @@ void test_l1_dsFPD_negative_dsGetFPColor (void)
  * |Step|Description|Test Data|Expected Result|Notes|
  * |:--:|-----------|---------|---------------|-----|
  * |01|Initialize using dsFPInit()||dsERR_NONE|Ensure the system is initialized|
- * |02|Call dsSetFPDMode() and loop through all valid values in dsFPDMode_t|eMode: [Valid Mode]|dsERR_NONE|API should set mode successfully|
- * |03|Terminate using dsFPTerm()||dsERR_NONE|Ensure the system is terminated|
+ * |02|Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_ON |dsERR_NONE|Ensure the system is initialized|
+ * |03|Call dsSetFPDMode() and loop through all valid values in dsFPDMode_t|eMode: [Valid Mode]|dsERR_NONE|API should set mode successfully|
+ * |04|Terminate using dsFPTerm()||dsERR_NONE|Ensure the system is terminated|
  *
  * @note This test case is deprecated. 
- * @note Valid indicators can retrieved from id element in kIndicators in the dsFPDSettings.h file
+ * @note Valid indicators can retrieved from yaml file
  */
 void test_l1_dsFPD_positive_dsSetFPDMode (void)
 {
     gTestID = 19;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     dsError_t result;
+    dsFPDIndicator_t eIndicator;
+    char buffer[DS_FPD_KEY_SIZE];
+    int count =0;
 
     // Step 01: Initialize using dsFPInit()
     result = dsFPInit();
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
+
+    // Step 02: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
+    count = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+    for (int i = 1; i <= count; i++)
+    {
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__,  eIndicator);
+        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
```

Comment:
This is never switched off.

Reply:
Done

Reply:
Please call FPD_STATE_OFF after 
  // Step 03: Call dsSetFPDMode() and loop through all valid values in dsFPDMode_t
    for (dsFPDMode_t mode = dsFPD_MODE_ANY; mode < dsFPD_MODE_MAX; ++mode)
    {
        result = dsSetFPDMode(mode);
        UT_ASSERT_EQUAL(result, dsERR_NONE);
    }

Reply:
Done